src/controllers/signAccountOp/signAccountOp.ts

```diff
@@ -2080,16 +2144,475 @@ export class SignAccountOpController extends EventEmitter implements ISignAccoun
+      } catch (e: any) {
+        let retryMsg
+
+        // if the signAccountOp is still active (it should be)
+        // try to switch the bundler and ask the user to try again
+        const switcher = this.bundlerSwitcher
+        this.updateStatus(SigningStatus.ReadyToSign)
+
+        if (switcher.canSwitch(baseAcc)) {
+          switcher.switch()
+          this.simulate()
+          this.gasPrice.fetch()
+          retryMsg = 'Broadcast failed because bundler was down. Please try again'
+        }
+
+        return this.throwBroadcastAccountOp({
+          error: e,
+          accountState,
+          provider: this.provider,
+          network: this.#network,
+          message: retryMsg
+        })
+      }
+      if (!userOperationHash) {
+        return this.throwBroadcastAccountOp({
+          message: 'Bundler broadcast failed. Please try broadcasting by an EOA or contact support.'
+        })
+      }
+
+      transactionRes = {
+        nonce: Number(userOperation.nonce),
+        identifiedBy: {
+          type: 'UserOperation',
+          identifier: userOperationHash,
+          bundler: bundler.getName()
+        }
+      }
+    }
+    // Smart account, the Relayer way
+    else {
+      try {
+        const body = {
+          gasLimit: Number(accountOp.gasFeePayment!.simulatedGasLimit),
+          txns: getSignableCalls(accountOp),
+          signature: accountOp.signature,
+          signer: { address: accountOp.signingKeyAddr },
+          nonce: Number(accountOp.nonce)
+        }
+        const additionalRelayerNetwork = relayerAdditionalNetworks.find(
+          (net) => net.chainId === this.#network.chainId
+        )
+        const relayerChainId = additionalRelayerNetwork
+          ? additionalRelayerNetwork.chainId
+          : accountOp.chainId
+        const response = await this.#callRelayer(
+          `/identity/${accountOp.accountAddr}/${relayerChainId}/submit`,
+          'POST',
+          body
+        )
+        if (!response.success) throw new Error(response.message)
+
+        transactionRes = {
+          txnId: response.txId,
+          nonce: Number(accountOp.nonce),
+          identifiedBy: {
+            type: 'Relayer',
+            identifier: response.id
+          }
+        }
+      } catch (error: any) {
+        return this.throwBroadcastAccountOp({ error, accountState, isRelayer: true })
+      }
+    }
+
+    if (!transactionRes)
+      return this.throwBroadcastAccountOp({
+        message: 'No transaction response received after being broadcasted.'
+      })
+
+    // simulate the swap & bridge only after a successful broadcast
+    if (this.#type === 'one-click-swap-and-bridge' || this.#type === 'one-click-transfer') {
+      this.portfolioSimulate().then(() => {
+        this.#portfolio.markSimulationAsBroadcasted(account.addr, this.#network.chainId)
+      })
+    } else {
+      this.#portfolio.markSimulationAsBroadcasted(account.addr, this.#network.chainId)
+    }
```

Comment:
The `if` from lines 2443 to 2449 can become an `onBroadcastSuccess` specific logic for main, transfer and swap as it would be more complete